build.sh

```diff
@@ -121,43 +121,41 @@ fi
 # Now we build everything.
 BUILD_DIR="$ROOT_DIR/build/"
 if [ ! -d "${BUILD_DIR}" ]; then
-mkdir -p "${BUILD_DIR}"
+  mkdir -p "${BUILD_DIR}"
 fi
 
 pushd "$BUILD_DIR"
 
 
-need_pickle5_backport=0
-PYTHON_REVISION="$("$PYTHON_EXECUTABLE" -s -c "import sys; print('{}.{}.{}'.format(*sys.version_info))")"
-case "${PYTHON_REVISION}.0" in
-3\.[5-7].*|3.8.[0-1].*) need_pickle5_backport=1;;
-esac
-
-if [ 0 -ne "${need_pickle5_backport}" ]; then
-WORK_DIR="$(mktemp -d)"
-pushd "${WORK_DIR}"
-git clone https://github.com/suquark/pickle5-backport
-pushd pickle5-backport
-  git checkout 8ffe41ceba9d5e2ce8a98190f6b3d2f3325e5a72
-  CC=gcc "$PYTHON_EXECUTABLE" setup.py --quiet bdist_wheel
-  unzip -q -o dist/*.whl -d "$ROOT_DIR/python/ray/pickle5_files"
-popd
-popd
-fi
-
-
-if [ -z "$SKIP_THIRDPARTY_INSTALL" ]; then
-    CC=gcc "$PYTHON_EXECUTABLE" -m pip install -q psutil setproctitle \
-            --target="$ROOT_DIR/python/ray/thirdparty_files"
-fi
-
-export PYTHON3_BIN_PATH="$PYTHON_EXECUTABLE"
-
 if [ "$RAY_BUILD_JAVA" == "YES" ]; then
   "$BAZEL_EXECUTABLE" build //java:ray_java_pkg --verbose_failures
 fi
 
 if [ "$RAY_BUILD_PYTHON" == "YES" ]; then
+  need_pickle5_backport=1
+  if PYTHONPATH="$ROOT_DIR/python/ray/pickle5_files:$PATHPATH" "$PYTHON_EXECUTABLE" -s -c "import pickle5" 2>/dev/null; then
```

Comment:
A couple things:

- If this should be here, could we separate the assignment from the conditional?

- I feel like `PYTHONPATH` shouldn't need to be modified here at all? I don't think it used to be before. Why is it needed now? Note that developers might want to just run things directly instead of running this script, and it's a bit of a pain to force them to change `PYTHONPATH` just to do that if it's not really necessary.

Reply:
hi @mehrdadn, this is not an assignment. the whole `PYTHONPATH="$ROOT_DIR/python/ray/pickle5_files:$PATHPATH" "$PYTHON_EXECUTABLE" -s -c "import pickle5" 2>/dev/null` is a command. and the `PYTHONPATH="$ROOT_DIR/python/ray/pickle5_files:$PATHPATH"` part is for setting env var for the following `python -s -c` command.

What this command does is checking whether `pickle5` is already available by call `import pickle5`. If it's unavailable, this command will fail with exit code != 0.

The reason why I set `PYTHONPATH` is because we install `pickle5` in that dir, see line 146.

Reply:
Oops, I somehow missed that there were multiple parts to the command and entirely misread the command, sorry! That makes sense, thanks!

Reply:
just updated code and comments.

Reply:
I think by `PATHPATH` you meant `PYTHONPATH`?

Reply:
yeah, fixed the typo